Analysis/src/Type.cpp

```diff
@@ -263,6 +264,20 @@ std::optional<TypeId> getMetatable(TypeId type, NotNull<BuiltinTypes> builtinTyp
 {
     type = follow(type);
 
+    if (FFlag::LuauIntersectedBinopOverloadFix)
+    {
+        if (const IntersectionType* itv = get<IntersectionType>(type))
+        {
+            for (TypeId part : itv->parts)
+            {
+                auto partMT = getMetatable(part, builtinTypes);
+                if (partMT != std::nullopt)
+                    return partMT;
+            }
```

Comment:
Appreciate the review! Thanks for the note on cyclic intersections and other edge cases. I definitely see the issue of handling intersections of multiple metatables and how that gets handled. That would need some refactoring outside of the scope of this PR here.

I'll look into the suggestions you mentioned in my free time, hope it can at least be an incremental step towards having fully sound intersection types and make it into production if it's all stable and doesn't make anything _worse_ per se. There was a related issue #983 involving the interaction between metatables and intersection types.